From 7748985cddfe59a6243fefce4a3f6d8547032ec5 Mon Sep 17 00:00:00 2001
From: XZS <d.f.fischer@web.de>
Date: Fri, 15 Jul 2016 13:56:11 +0200
Subject: [PATCH] support socket activation through systemd

When run as a standalone server, it can be helpful to have vinefeed
activated on demand.
---
 bin/www      | 2 +-
 package.json | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/bin/www b/bin/www
index 06a579f..58016fb 100755
--- a/bin/www
+++ b/bin/www
@@ -13,5 +13,5 @@ app.set 'port', process.env.PORT

 console.log "Environment: #{app.get('env')}"

-server = app.listen app.get('port'), ->
+server = app.listen require('systemd-socket')() || app.get('port'), ->
 	console.log "Listening on port #{server.address().port}"
diff --git a/package.json b/package.json
index 7506238..43a19b5 100644
--- a/package.json
+++ b/package.json
@@ -9,6 +9,7 @@
     "feed": "^0.2.6",
     "minimist": "^1.1.0",
     "morgan": "^1.3.2",
-    "request": "^2.45.0"
+    "request": "^2.45.0",
+    "systemd-socket": "^0.0.0"
   }
 }
--
2.9.0
